src/Manifest/ManifestReader.php

```diff
@@ -85,10 +85,15 @@ public function readManifests(array $moduleTransfers, IntegratorCommandArguments
                 continue;
             }
 
-            $manifest = json_decode($json, true);
-
-            if ($manifest) {
-                $manifests[$moduleFullName] = $manifest;
+            $manifestFileData = json_decode($json, true);
+            if ($manifestFileData) {
+                foreach ($manifestFileData as &$strategy) {
+                    foreach ($strategy as &$manifest) {
+                        $manifest[IntegratorConfig::MODULE_KEY] = $moduleFullName;
+                        $manifest[IntegratorConfig::MODULE_VERSION_KEY] = $version;
+                    }
+                }
+                $manifests[$moduleFullName] = $manifestFileData;
```

Comment:
You have to destroy references after foreach loop
```suggestion
                unset($strategy, $manifest);
                $manifests[$moduleFullName] = $manifestFileData;
```

Reply:
++